examples/src/main/java/com/mapbox/navigation/examples/core/AdasisActivity.kt

```diff
@@ -0,0 +1,325 @@
+    private val options: MapboxRouteLineOptions by lazy {
+        MapboxRouteLineOptions.Builder(this)
+            .withRouteLineResources(routeLineResources)
+            .withRouteLineBelowLayerId("road-label")
+            .build()
+    }
+
+    private val routeLineView by lazy {
+        MapboxRouteLineView(options)
+    }
+
+    private val routeLineApi: MapboxRouteLineApi by lazy {
+        MapboxRouteLineApi(options)
+    }
+
+    private val routeArrowApi: MapboxRouteArrowApi by lazy {
+        MapboxRouteArrowApi()
+    }
+
+    private val routeArrowView: MapboxRouteArrowView by lazy {
+        MapboxRouteArrowView(RouteArrowOptions.Builder(this).build())
+    }
+
+    private val replayProgressObserver = ReplayProgressObserver(mapboxReplayer)
+
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        routeArrowApi.addUpcomingManeuverArrow(routeProgress).apply {
+            routeArrowView.renderManeuverUpdate(mapboxMap.getStyle()!!, this)
+        }
+    }
+
+    private val locationObserver = object : LocationObserver {
+        override fun onNewRawLocation(rawLocation: Location) {}
+        override fun onNewLocationMatcherResult(locationMatcherResult: LocationMatcherResult) {
+            navigationLocationProvider.changePosition(
+                locationMatcherResult.enhancedLocation,
+                locationMatcherResult.keyPoints,
+            )
+            if (isNavigating) {
+                updateCamera(locationMatcherResult.enhancedLocation)
+            }
+        }
+    }
+
+    private val routesObserver = RoutesObserver { result ->
+        if (result.navigationRoutes.isNotEmpty()) {
+            CoroutineScope(Dispatchers.Main).launch {
+                routeLineApi.setNavigationRoutes(result.navigationRoutes).apply {
+                    routeLineView.renderRouteDrawData(mapboxMap.getStyle()!!, this)
+                }
+            }
+            isNavigating = true
+            startSimulation(result.navigationRoutes[0].directionsRoute)
+        }
+    }
+
+    private var adasisMessagesLog: String = "Received adasis messages:\n"
+    private var numberOfMessages = 1
+
+    private fun init() {
+        initNavigation()
+        initStyle()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initNavigation() {
+        mapboxNavigation = MapboxNavigationProvider.create(
+            NavigationOptions.Builder(this@AdasisActivity)
+                .accessToken(getMapboxAccessTokenFromResources())
+                .locationEngine(ReplayLocationEngine(mapboxReplayer))
+                .build()
+        )
+        mapboxNavigation.registerLocationObserver(object : LocationObserver {
+            override fun onNewRawLocation(rawLocation: Location) {
+                updateCamera(rawLocation)
+                navigationLocationProvider.changePosition(
+                    rawLocation,
+                )
+                mapboxNavigation.unregisterLocationObserver(this)
```

Comment:
Why do you unregister it?

Reply:
We need to move camera only on the first update. This is typical code from all our example activities